packages/apps/desktop-viewer-test/src/frontend/hooks/usePullChanges.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { BriefcaseConnection } from "@itwin/core-frontend";
+import type { ProgressCallback } from "@itwin/core-frontend/lib/cjs/request/Request";
+import { useCallback, useState } from "react";
+
+export const usePullChanges = (connection?: BriefcaseConnection) => {
+  const [pullProgress, setPullProgress] = useState<number>();
+
+  const progressCallback: ProgressCallback = useCallback((progress) => {
+    const { loaded, total, percent } = progress;
+
+    const progressPercentage = total ? (loaded / total) * 100 : percent;
+    setPullProgress(progressPercentage);
+
+    console.log(
+      `Pull changes progress (${loaded}/${total}) -> ${progressPercentage}%`
```

Comment:
> do we need to log loaded/total?

Would be more consistent if we did since we do log it with briefcase download ([useDownload.tsx](https://github.com/iTwin/viewer/blob/919ffb9f2d467035cecdc0a74c3259f08e20b020/packages/apps/desktop-viewer-test/src/frontend/hooks/useDownload.tsx#L41)).

> you check earlier if total is defined, if its not its going to log undefined which will look weird

The way progress is defined in the core, total will be always defined and percentage will always be undefined. This and the fact that `ProgressCallback` is internal, kinda screams that `ProgressCallback` shouldn't be used neither here nor with briefcase download.

Reply:
Opened https://github.com/iTwin/itwinjs-core/pull/4899 to address this is issue.